File: Rust/0050-powx-n.rs
impl Solution {
    pub fn my_pow(mut x: f64, n: i32) -> f64 {
        if n == 0 || x == 1.0  {
            return 1.0;
        } else {
            let mut n_abs: i64 = n.into();
            let mut result: f64 = 1.0;
        
            if(n < 0) {
                n_abs = -n_abs;
            }

            while(n_abs != 0) {
                if(n_abs % 2 == 0) {
                    x = x * x;
                    n_abs = n_abs / 2;
                } else {
                    result = result * x;
                    n_abs = n_abs - 1;
                }
            }
        
            return if (n < 0) { 1.0 / result } else { result };
        }
    }
}

File: dev-c++/0050-powx-n.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if(n == 0 || x == 1.0) { return 1.0; }

        long n_abs = n;
        double result = 1.0;

        if(n < 0) {
            n_abs = -n_abs;
            x = 1 / x;
        }   

        while(n_abs != 0) {
            if(n_abs % 2 == 0) {
                x *= x;
                n_abs /= 2;
            } else {
                result *= x;
                n_abs -= 1;
            }
        }

        return result;
    }
};

